Ingester Panic with full disk

**Describe the bug**
We have seen the following ingester panic after the PVC has filled:

```
panic: runtime error: slice bounds out of range [:1717988153] with capacity 875389745
goroutine 462 [running]:
github.com/grafana/tempo/tempodb/encoding/base.UnmarshalObjectFromReader(0x2626240, 0xc0005b08b8, 0xc05ddea010, 0xe5f0, 0xe5f0, 0x0, 0x0, 0xe5f0, 0xc000cce9c0, 0x23973d0)
	/drone/src/tempodb/encoding/base/object.go:70 +0x494
github.com/grafana/tempo/tempodb/encoding.(*iterator).Next(0xc000928270, 0x26618b8, 0xc000136018, 0xc06e9feac0, 0x10, 0x10, 0xc05ddea010, 0xe5f0, 0xe5f0, 0x0, ...)
	/drone/src/tempodb/encoding/iterator.go:30 +0x3b
github.com/grafana/tempo/modules/ingester.(*Ingester).replayBlock(0xc000249600, 0xc000cc0690, 0xc000cce9c0, 0x0, 0x0)
	/drone/src/modules/ingester/ingester.go:321 +0xeb
github.com/grafana/tempo/modules/ingester.(*Ingester).replayWal(0xc000249600, 0x7f13420b6658, 0x10)
	/drone/src/modules/ingester/ingester.go:298 +0x3dc
github.com/grafana/tempo/modules/ingester.(*Ingester).starting(0xc000249600, 0x2661880, 0xc000a86e00, 0xc0005a3b01, 0xc0005a3bf0)
	/drone/src/modules/ingester/ingester.go:92 +0x2f
github.com/cortexproject/cortex/pkg/util/services.(*BasicService).main(0xc04114e5a0)
	/drone/src/vendor/github.com/cortexproject/cortex/pkg/util/services/basic_service.go:134 +0x45e
created by github.com/cortexproject/cortex/pkg/util/services.(*BasicService).StartAsync.func1
	/drone/src/vendor/github.com/cortexproject/cortex/pkg/util/services/basic_service.go:96 +0xad
```

Full disk is an exceptional situation, but the ingester should handle it more gracefully. Review options to avoid this panic.

--- a/tempodb/encoding/base/object.go
+++ b/tempodb/encoding/base/object.go
@@ -66,6 +66,9 @@ func UnmarshalObjectFromReader(r io.Reader) (common.ID, []byte, error) {
 	if uint32(readLength) != protoLength {
 		return nil, nil, fmt.Errorf("read %d but expected %d", readLength, protoLength)
 	}
+	if int(idLength) > len(b) {
+		return nil, nil, fmt.Errorf("id length %d outside bounds of buffer %d. corrupt buffer?", idLength, len(b))
+	}
 
 	bytesID := b[:idLength]
 	bytesObject := b[idLength:]

